[rocm] Nightly pytorch + rocm test failures

Sporadic failures running `pytest` on main with

```
pytorch-triton-rocm             3.5.0+git0795ae34.rocm7.10.0a20251024
torch                           2.10.0a0+rocm7.10.0a20251024
torchaudio                      2.10.0a0+rocm7.10.0a20251024
torchvision                     0.25.0a0+rocm7.10.0a20251024
rocm                            7.10.0a20251024
rocm-sdk-core                   7.10.0a20251024
rocm-sdk-libraries-gfx94X-dcgpu 7.10.0a20251024
```

E.g.,

```
FAILED tests/dynamo/backend_test.py::test_generic_backend_backward[device1] - torch.AcceleratorError: HIP error: peer access is already enabled
FAILED tests/runtime/device_stream_test.py::DeviceStreamTest::testEagerCustomOpMultiStream - torch.AcceleratorError: HIP error: peer access is already enabled
FAILED tests/runtime/launch_test.py::LaunchableTest_1::testLaunchJit - RuntimeError: Comparing
================================================== 3 failed, 495 passed, 3 skipped, 8 xfailed, 11 warnings in 171.52s (0:02:51) ==================================================
python: iree/runtime/src/iree/hal/drivers/hip/hip_allocator.c:81: iree_hal_hip_allocator_t *iree_hal_hip_allocator_cast(iree_hal_allocator_t *): Assertion `IREE_UNLIKELY(!!(iree_hal_resource_is(base_value, &iree_hal_hip_allocator_vtable)))' failed.
Aborted (core dumped)
```

It's a bit hard to debug, but I expect that somewhere in pytorch+rocm code there is an unguarded call to `hipDeviceEnablePeerAccess`, which is getting called after IREE has already enabled this. But frankly, I'm not sure.

--- a/runtime/src/iree/hal/drivers/hip/dynamic_symbol_tables.h
+++ b/runtime/src/iree/hal/drivers/hip/dynamic_symbol_tables.h
@@ -47,6 +47,7 @@ IREE_HAL_HIP_OPTIONAL_PFN_DECL(hipGetDeviceProperties, hipDeviceProp_tR0000*,
 // const char* instead of hipError_t so it uses a different macro.
 IREE_HAL_HIP_REQUIRED_PFN_STR_DECL(hipGetErrorName, hipError_t)
 IREE_HAL_HIP_REQUIRED_PFN_STR_DECL(hipGetErrorString, hipError_t)
+IREE_HAL_HIP_REQUIRED_PFN_DECL(hipGetLastError)
 IREE_HAL_HIP_REQUIRED_PFN_DECL(hipGraphAddEmptyNode, hipGraphNode_t*,
                                hipGraph_t, const hipGraphNode_t*, size_t)
 IREE_HAL_HIP_REQUIRED_PFN_DECL(hipGraphAddEventRecordNode, hipGraphNode_t*,

--- a/runtime/src/iree/hal/drivers/hip/hip_device.c
+++ b/runtime/src/iree/hal/drivers/hip/hip_device.c
@@ -453,7 +453,9 @@ static iree_status_t iree_hal_hip_device_enable_peering(
 
     hip_error = symbols->hipDeviceEnablePeerAccess(j, 0);
     if (hip_error == hipErrorPeerAccessAlreadyEnabled) {
-      // Already peered. That's okay.
+      // Already peered. That's okay. Clear the sticky error so it doesn't
+      // affect subsequent HIP calls.
+      symbols->hipGetLastError();
       continue;
     } else if (hip_error != hipSuccess) {
       // Unhandled error, propagate it.

